### ISSUE TITLE
Handle async/generator functions in native way

### ISSUE THREAD

**Issue by Strate** [mobxjs/babel-plugin-mobx-deep-action#2]
Plugin should handle async and generator functions natively, without dependency of `transform-regenetator`. It should search & wrap in actions async parts of code. Examples:

```js
action(async function handler() {
  this.a = await other();
  this.b = await other();
  this.state = [a,b];
})
```
should be transformed to:
```js
action(async function handler() {
  let result_0 = await other();
  return await (action(async function inner_1() {
    this.a = result_0;
    let result_0 = await other();
    return await (action(async function inner_2() {
      this.b = result_0;
      this.state = [a,b];
    }))()
  }))();
})
```
This is pretty hard to implement, and it is reasonable only for environments with native async/generators support. Without that support this code still have to be transformed via `transform-regenerator` plugin, which become more complex and expensive, because a lot of inner async/generator functions.

**Comment by andrew-luminal** [comment]
👍 Would love this functionality as well!

**Comment by cheft** [comment]
Can you provide an example?

**Comment by Strate** [comment]
I think we can transform native async/await code to asyncAction from mobx-utils: https://github.com/mobxjs/mobx-utils#asyncaction

**Comment by Strate** [comment]
Closed in favor of #12
